openmls/src/messages/tests/test_welcome.rs

```diff
@@ -162,26 +163,24 @@ fn test_welcome_ciphersuite_mismatch(
     let group_info_bytes = welcome_key
         .aead_open(backend, welcome.encrypted_group_info(), &[], &welcome_nonce)
         .expect("Could not decrypt GroupInfo.");
-    let mut group_info = GroupInfo::tls_deserialize(&mut group_info_bytes.as_slice())
-        .expect("Could not deserialize GroupInfo.");
+    let mut verifiable_group_info =
+        VerifiableGroupInfo::tls_deserialize(&mut group_info_bytes.as_slice()).unwrap();
 
     // Manipulate the ciphersuite in the GroupInfo
-    group_info
+    verifiable_group_info
         .payload
         .group_context
         .set_ciphersuite(mismatched_ciphersuite);
 
     // === Reconstruct the Welcome message and try to process it ===
 
-    let group_info_bytes = group_info
-        .tls_serialize_detached()
-        .expect("Could not serialize GroupInfo.");
+    let verifiable_group_info_bytes = verifiable_group_info.tls_serialize_detached().unwrap();
 
-    let encrypted_group_info = welcome_key
-        .aead_seal(backend, &group_info_bytes, &[], &welcome_nonce)
-        .expect("Could not encrypt GroupInfo.");
+    let encrypted_verifiable_group_info = welcome_key
+        .aead_seal(backend, &verifiable_group_info_bytes, &[], &welcome_nonce)
+        .unwrap();
 
-    welcome.encrypted_group_info = encrypted_group_info.into();
+    welcome.encrypted_group_info = encrypted_verifiable_group_info.into();
```

Comment:
Here, we "coerce" a previously *unverified* `group info (seemingly) into a group info (`encrypted_group_info` vs `unverified_encrypted_group_info`). I'm not sure if this is something we should worry about right now.

Reply:
What we want would be something like: Encryption only for `GroupInfo`, decryption should always yield a `UnverifiedGroupInfo`. For tests, I think coercion is okay.